kani-driver/src/main.rs

```diff
@@ -73,6 +77,23 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
         let result = ctx.check_harness(&specialized_obj, &report_dir, harness)?;
         if result == VerificationStatus::Failure {
             failed_harnesses.push(harness);
+
+            if !has_run_once && ctx.args.gen_exec_trace {
```

Comment:
Can you move this code to be outside of this loop? Instead of having the `has_run_once`, you can just iterate over the list of `failed_harnesses`. For now, the case where you only run this once, you can just get the first element of the list.